### ISSUE TITLE
Uncaught DOMException: Failed to execute 'setAttribute' on 'Element': '' is not a valid attribute name

### ISSUE THREAD

**Issue by ghost** [ractivejs/ractive#2783]
## Description:
Chrome console is spitting out this error:

``Uncaught DOMException: Failed to execute 'setAttribute' on 'Element': '' is not a valid attribute name``

I have a hard time reproducing this. Seems as if Ractive is re-rendering an attribute that on the second render is no longer availible.

Line 9805 of ractive.js:
```
  function updateAttribute ( reset ) {
  	if ( reset ) this.node.removeAttribute( this.name );
  	else this.node.setAttribute( this.name, safeToStringValue( this.getString() ) );
  }
```
I've done a console log here and get blank `this.name` and `this.getString()`.

By doing the following it works without any problem:

```
  function updateAttribute ( reset ) {
  	if ( reset ) this.node.removeAttribute( this.name );
  	else {
      if(this.name)
        this.node.setAttribute( this.name, safeToStringValue( this.getString() ) );
    }
  }
```
This is a ok patch fix for now. Must be something deeper that is causing this.

## Versions affected:
v0.8.1

## Platforms affected:
Chrome

## **Code:**
```
                        <div class="field dropdown"

                          {{#if i == 0}}
                            style="width: {{~/duplicateTableSettings.fixedCellSize}}px;"
                          {{elseif ~/duplicateTableSettings.columnWidth > 0}}
                            style="width: {{~/duplicateTableSettings.columnWidth}}px;"
                          {{/if}}

                          {{#fields[.].length > 1 && !dropdownVisible("showAllFieldValues", group[groupIndex].id + "_field_" + .)}}
                            data-balloon="{{fields[.].length - 1}} more"
                          {{elseif fields[.].length <= 1}}
                            {{#if fields[.][0].badInput}}
                              data-balloon="Invalid {{fields[.][0].badInput}}"
                            {{/if}}
                          {{/if}}

                          data-balloon-pos="down"
                        >
```

**Comment by ghost** [comment]
While everyone digs into this issue, can you verify if the latest (0.8.5) fixes this? It might have been fixed. 😄 

It's also a sign of a bigger problem. Your code doesn't appear to have invalid attributes (there doesn't appear to be any, but I might be blind at the moment 😉 ), but `name` becomes blank. It might be an error in how the attributes are parsed, leading to that blank. A check on `name` might not be the best solution.

I would be nice to have a reproduction on a fiddle.

If it's really hard to reproduce, can we at least have the instance's definition and the current data on the instance (I believe the cool kids call it "state" nowadays). You can use `ractive.get()` to extract the current data.

If all other options fail, a stack trace leading up to the error, and the scope data would be nice too. While it might not help us know how `this.name` became blank, at least we know the flow of the code leading up to the error and infer what operations caused it.

**Comment by ghost** [comment]
Well, I'll eventually dig deeper since I want to keep updated on new Ractive versions. Just checked the new 0.8.5 and it still has the same problems. I'll stick with my fork for the moment and once it gets too out of date I'll jump back on this.

I'm not sure how to catch the state in between attribute re-renders. I suppose I could inject a ractive.get in the code for a messy debug.

**Comment by ghost** [comment]
I think I narrowed it down after realizing that `name` is something the parser results can tell me without having a running setup. https://jsfiddle.net/gsbqmt9a/

```html
<script type="text/ractive" id="broken">
<div
  {{#fields[.].length > 1 && !dropdownVisible("showAllFieldValues", group[groupIndex].id + "_field_" + .)}}
    data-balloon="{{fields[.].length - 1}} more"
  {{elseif fields[.].length <= 1}}
    {{#if fields[.][0].badInput}}
      data-balloon="Invalid {{fields[.][0].badInput}}"
    {{/if}}
  {{/if}}
></div>
</script>

<script type="text/ractive" id="not-broken">
<div
  {{#fields[.].length > 1 && !dropdownVisible("showAllFieldValues", group[groupIndex].id + "_field_" + .)}}
    data-balloon="{{fields[.].length - 1}} more"
  {{elseif fields[.].length <= 1}}{{#if fields[.][0].badInput}}
      data-balloon="Invalid {{fields[.][0].badInput}}"
  {{/if}}{{/if}}
></div>
</script>
```

The problem appears to be the newlines in the `elseif` section. The parser results seem to indicate that it is treating the newlines outside the `{{#if}}` section (both before and after) as "blank attributes". It's probably the only time `setAttribute` chokes.

**Broken**: Notice the two empty attribute entries before and after the `data-balloon` attribute entry.
![broken](https://cloud.githubusercontent.com/assets/1582932/20955210/4e63034c-bc0f-11e6-822a-dff3d39eac21.png)

**Not broken**: Removing the newline makes them go away
![not-broken](https://cloud.githubusercontent.com/assets/1582932/20955222/711b8562-bc0f-11e6-92ea-ac73ccf41cdd.png)

A potential workaround until this gets fixed, without patching Ractive, would be to get rid of the inner `{{#if}}` and just merge the condition with the one in the `{{elseif}}`. https://jsfiddle.net/gsbqmt9a/1/

```html
<script type="text/ractive" id="workaround">
<div
  {{#fields[.].length > 1 && !dropdownVisible("showAllFieldValues", group[groupIndex].id + "_field_" + .)}}
    data-balloon="{{fields[.].length - 1}} more"
  {{elseif fields[.].length <= 1 && fields[.][0].badInput}}
      data-balloon="Invalid {{fields[.][0].badInput}}"
  {{/if}}
></div>
</script>
```

**Comment by evs-chris** [comment]
Nice sleuthing! This is now fixed in edge and v0.8-dev, and it will go out with 0.8.8 when it gets cut.

**Comment by ghost** [comment]
Thanks a million! I'll try the workaround until 0.8.8.
